src/SqlAsyncCollector.cs

```diff
@@ -313,6 +313,11 @@ private static void GenerateDataQueryForMerge(TableInformation table, IEnumerabl
                 else
                 {
                     // ToDo: add check for duplicate primary keys once we find a way to get primary keys.
+                    // lower case the json keys if sql table is case insensitive since the query we built expects everything lowercased.
+                    if (table.Comparer == StringComparer.OrdinalIgnoreCase)
+                    {
+                        ChangePropertiesToLowerCase(row as JObject);
```

Comment:
So we already have JsonSerializerSettings in table information - which we then use to generate the rowData. Why isn't that handling converting it to lowercase? It would be good to look into using that same logic instead of doing this custom logic here.

Reply:
I already tried that before coming to this solution, it was not changing the case and I was not able to find out why.

Reply:
Alright, so I looked into this and the root issue here is because contract resolvers are ignored by default when serializing JObjects (https://web.archive.org/web/20171005181503/http://json.codeplex.com/workitem/23853).

I looked around and unfortunately there doesn't seem to be a clean way to change this behavior - so manually going through and updating the property names like you have is probably our best solution.